### DISCUSSION TITLE
How to configure null-ls to increase the timeout limits?

### DISCUSSION THREAD

**Discussion by jarmos-san** [LunarVim/LunarVim#2767] [Q&A] [answered]
I'm facing issues with ESLint & Prettier taking a little more than a couple of seconds to format/lint the buffer, hence, Neovim keeps timing out. Is there any way to increase the timeout limits without messing around with the default LSP configurations LunarVim provides?

**Comment by b4rlw** [comment]
I'm facing the same issue with Black for Python. I saw `vim.lsp.buf.format({ timeout_ms = 5000 })` as a solution elsewhere, but unfortunately this doesn't seem to be valid input for Lunarvim's `config.lua`.

**Reply by b4rlw** [reply]
You can try:
`lvim.lsp.null_ls.setup.timeout_ms = 5000`
I'll need to do some more testing before I can say if it worked for my case, but it may help here.

**Reply by jarmos-san** [reply]
If you can find a workaround for this? Please share it here so that I can mark it as a valid answer.

**Reply by manasomali** [reply]
up, same issue here... can't find the solution for this anywhere.

**Comment by maxdinech** [comment]
This is quite annoying with black on Python files... I hope there is an answer soon.

**Reply by jarmos-san** [reply]
I ended up using [AstroNvim](https://github.com/AstroNvim/AstroNvim) because it provides a configuration framework much closer to what you normally would do if you chose to configure Neovim from scratch. Astronvim also doesn't wrap the `vim` API which means it's way more intuitive & you can simply refer to the Neovim docs when needed.

**Reply by galexbh** [reply]
how did you make this configuration in astro vim?

**Reply by jarmos-san** [reply]
I don't remember the exact co configuration but I believe null-ls discusses the issue in its docs so check it out. Besides I now resorted to using prettierd instead which is so much more responsive! See this line of code - https://github.com/Jarmos-san/dotfiles/blob/c022eb586fa7e4c609ccb5b1659928d073a6e2a6/dotfiles/private_dot_config/nvim/lua/user/plugins/null-ls.lua#L7

**Reply by mohit2152sharma** [reply]
for astrovim, you can add the following in your `init.lua` file:

```lua
return {
   lsp = {
      formatting = {
         timeout_ms = 3200
         }
     }
}
```

**Comment by kylo252** [comment, marked as answer]
You can simply override the default `<leader>lf` binding to have a longer timeout value
```lua
lvim.builtin.which_key.mappings["l"]["f"] = {
  function()
    vim.lsp.format { timeout = 2000 }
  end,
  "LSP format",
}
```

**Reply by lzcn** [reply]
For the master branch, I modified your code as follows:
```lua
lvim.builtin.which_key.mappings["l"]["f"] = {
  function()
    require("lvim.lsp.utils").format { timeout_ms = 2000 }
  end,
  "Format",
}
```

**Reply by NiklasKappel** [reply]
Just for completeness because I keep coming back to this thread, to format reliably on save you can use:
```
lvim.format_on_save.enabled = true
lvim.format_on_save.timeout = 5000
```
